models/ast/ast_node_evaluation.go

```diff
@@ -35,24 +35,11 @@ func (root NodeEvaluation) AllErrors() (errs []error) {
 	return errs
 }
 
-type RootNodeEvaluation struct {
-	ReturnValue bool
-	Errors      []error
-
-	Children      []NodeEvaluation
-	NamedChildren map[string]NodeEvaluation
-}
-
-func AdaptRootNodeEvaluation(root NodeEvaluation) (RootNodeEvaluation, error) {
```

Comment:
Roll back because it's anoying to use inside recursive function (need to handle recurisve func for `NodeEvaluation` due to chil/named chil and a "root" one for `RootNodeEvaluation`)